cli/src/lib.rs

```diff
@@ -1064,10 +1112,18 @@ fn init(
     )?;
 
     if !no_install {
-        let yarn_result = install_node_modules("yarn")?;
-        if !yarn_result.status.success() {
-            println!("Failed yarn install will attempt to npm install");
-            install_node_modules("npm")?;
+        if package_manager_cmd != PackageManager::NPM.to_string() {
+            let package_manager_result = install_node_modules(&package_manager_cmd)?;
+
+            if !package_manager_result.status.success() {
+                println!(
+                    "Failed {} install will attempt to npm install",
+                    package_manager_cmd
+                );
+                install_node_modules("npm")?;
+            }
+        } else {
+            install_node_modules(&package_manager_cmd)?;
```

Comment:
Nit: The above `if` check can be moved down to the inner `if` check (combined with `&&`) to get rid of the `else` code below.